### PR TITLE
Define module for managing E-Series email alerts

### PR DESCRIPTION
**Pull request by lmprice**
##### SUMMARY
Email alerts can be enabled for an E-Series system to provide
information to interested users by email when a warning or critical
level event occurs on the system. This module will allow a system owner
to configure whether or not system alerts are enabled, and who will
receive them.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME
netapp_e_alerts

##### ANSIBLE VERSION

```
ansible 2.6.0 (detached HEAD ce8c336d52) last updated 2018/04/18 11:23:03 (GMT -500)
  config file = None
  configured module search path = [u'/home/lorenp/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/lorenp/ansible/lib/ansible
  executable location = /home/lorenp/ansible/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Review by ndswartz — APPROVED** [review]
shipit

**Comment by mattclay** [comment]
The sanity failure is unrelated and has been fixed. I've restarted CI.

File: lib/ansible/modules/storage/netapp/netapp_e_alerts.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+author: Michael Price (@lmprice)
+extends_documentation_fragment:
+    - netapp.eseries
+options:
+    state:
+        description:
+            - Enable/disable the sending of email-based alerts.
+        default: enabled
+        required: false
+        choices:
```
Comment (gundalow) [inline lib/ansible/modules/storage/netapp/netapp_e_alerts.py:31]: Should this just be enabled/disabled?

What's present/absent mean?

File: lib/ansible/modules/storage/netapp/netapp_e_alerts.py [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+            - present
+            - absent
+            - enabled
+            - disabled
+    server:
+        description:
+            - A fully qualified domain name, IPv4 address, or IPv6 address of a mail server.
+            - To use a fully qualified domain name, you must configure a DNS server on both controllers using
+             M(netapp_e_mgmt_interface).
+             - Required when I(state=enabled/present).
```
Comment (gundalow) [inline lib/ansible/modules/storage/netapp/netapp_e_alerts.py]: I think it's confusing having enabled and present

File: lib/ansible/modules/storage/netapp/netapp_e_alerts.py [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+    test:
+        description:
+            - When a change is detected in the configuration, a test email will be sent.
+            - This may take a few minutes to process.
+            - Only applicable if I(state=enabled/present).
+        default: no
+        type: bool
+    log_path:
+        description:
+            - A local path to a file to be used for debug logging
```
Comment (gundalow) [inline lib/ansible/modules/storage/netapp/netapp_e_alerts.py]: Local to what?
The Ansible Controller?

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/storage/netapp/netapp_e_alerts.py (hunk 2 above)
Comment (lmprice) [reply lib/ansible/modules/storage/netapp/netapp_e_alerts.py]: The idea was to have consistency with all of the other places that state is utilized in Ansible modules. state=present/absent seems to be a common convention elsewhere (hence why prior developers used it in our volume module, for instance).

Comment (gundalow) [reply lib/ansible/modules/storage/netapp/netapp_e_alerts.py]: Sure, though the issue is you are supporting both present and enabled. What's the difference?
Just pick one pair and use that

Comment (gundalow) [reply lib/ansible/modules/storage/netapp/netapp_e_alerts.py]: enabled/disabled for this as we are talking about a service (email)

File: lib/ansible/utils/module_docs_fragments/netapp.py [hunk 4]
```diff
@@ -136,24 +136,31 @@ class ModuleDocFragment(object):
...
     default: true
     description:
-    - Should https certificates be validated?
+        - Should https certificates be validated?
+    type: bool
   ssid:
     required: true
     description:
     - The ID of the array to manage. This value must be unique for each array.
+
```
Comment (gundalow) [inline lib/ansible/utils/module_docs_fragments/netapp.py]: If this needs updating do it in it's own PR

**Comment by lmprice** [comment]
@gundalow The requested changes have been completed. Let me know if I missed something.

**Review by gundalow — APPROVED** [review]
